guide/src/faq.md

```diff
@@ -28,6 +28,22 @@ extension-module = ["pyo3/extension-module"]
 default = ["extension-module"]
 ```
 
+## I can't run `cargo test`: Compiler fails to import crate for tests in `tests/` directory!
+
+The Rust book suggests to [put integration tests inside a `tests/` directory](https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests). However, you can't do that for a PyO3 project. The compiler wouldn't be able to find your crate and displays an error like this:
+
+```
+error[E0463]: can't find crate for `my-crate`
+   --> tests/tests.rs:4:1
+    |
+  4 | use crate;
+    | ^^^^^^^^^ can't find crate
+```
+
+This is not a bug in PyO3. But it's a [known problem of Cargo](https://github.com/rust-lang/cargo/issues/6659).
+
+For now you can work around it by putting the integration tests inside the `src/` directory.
```

Comment:
I think an alternative workaround is also to add `rlib` to the `crate-types` in your `Cargo.toml`?

Reply:
Agreed.

Reply:
That is also what a colleague suggested. However that yielded a new error:

``` 
  = note: /usr/bin/ld: /home/developer/projects/pyo3-ftp/target/debug/deps/libpyo3-9f95204c15f7bad5.rlib(ceval.o): relocation R_X86_64_PC32 against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
          /usr/bin/ld: final link failed: Bad value
          collect2: error: ld returned 1 exit status
``` 

Any idea if this could that also happen to others? Or is this problem isolated for my project?

Reply:
Ah, I understand, thanks.
It looks like that we cannot link Python with rlib.
How about lib?

Reply:
GLIBC 2.2.5 is apparently from 2002... what OS are you using / where did you get the python interpreter from?

https://sourceware.org/glibc/wiki/Glibc%20Timeline